sdk/python/kfp/compiler/compiler.py

```diff
@@ -683,44 +683,15 @@ def _validate_exit_handler_helper(group, exiting_op_names, handler_exists):
 
     return _validate_exit_handler_helper(pipeline.groups[0], [], False)
 
-  def _compile(self, pipeline_func):
-    """Compile the given pipeline function into workflow."""
-
-    argspec = inspect.getfullargspec(pipeline_func)
-
-    # Create the arg list with no default values and call pipeline function.
-    # Assign type information to the PipelineParam
-    pipeline_meta = _extract_pipeline_metadata(pipeline_func)
-    pipeline_name = K8sHelper.sanitize_k8s_name(pipeline_meta.name)
-
-    args_list = []
-    for arg_name in argspec.args:
-      arg_type = None
-      for input in pipeline_meta.inputs:
-        if arg_name == input.name:
-          arg_type = input.param_type
-          break
-      args_list.append(dsl.PipelineParam(K8sHelper.sanitize_k8s_name(arg_name), param_type=arg_type))
-
-    with dsl.Pipeline(pipeline_name) as p:
-      pipeline_func(*args_list)
-
-    # Remove when argo supports local exit handler.
-    self._validate_exit_handler(p)
-
-    # Fill in the default values.
-    args_list_with_defaults = [dsl.PipelineParam(K8sHelper.sanitize_k8s_name(arg_name))
-                               for arg_name in argspec.args]
-    if argspec.defaults:
-      for arg, default in zip(reversed(args_list_with_defaults), reversed(argspec.defaults)):
-        arg.value = default.value if isinstance(default, dsl.PipelineParam) else default
+  def _postprocess(self, pipeline: dsl.Pipeline):
```

Comment:
instead of naming is postprocess,which is very generic, I recommend using a different meaningful name that explains a bit about what is happening. inject_artifact_location?

Reply:
Renamed it as _sanitize_and_inject_artifact, which is what it does.